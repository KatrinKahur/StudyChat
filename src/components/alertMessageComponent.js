import {Pressable, Text} from "react-native";
import {Feather} from "@expo/vector-icons";

export const AlertMessageComponent = ({message, chosenStyle = "info", setMessageCallback}) => {
    if(message === null){
        return (<></>)
    }

    if(chosenStyle !== "danger" && chosenStyle !== "success")
        chosenStyle = "info";

    return (
        <Pressable
            testID="message-container"
            onPress={() => setMessageCallback(null)}
            style={{...styles.box.base, ...styles.box[chosenStyle]}}>
            <Text style={{...styles.text.base, ...styles.text[chosenStyle]}}>{message}
                <Feather name="x" size={styles.text.base.fontSize} color={styles.text[chosenStyle].color} /></Text>
        </Pressable>
    )
}

const styles = {
    text: {
        info: {
            color: "#31708f"
        },
        danger: {
            color: "#a94442"
        },
        success: {
            color: "#3c763d"
        },
        base: {
            fontSize: 20
        }
    },
    box: {
        info: {
            backgroundColor: "#d9edf7",
            borderColor: "#bce8f1"
        },
        danger: {
            backgroundColor: "#f2dede",
            borderColor: "#ebccd1"
        },
        success: {
            backgroundColor: "#dff0d8",
            borderColor: "#d6e9c6"
        },
        base: {
            zIndex: 9999,
            position: "absolute",
            top: 10,
            borderWidth: 1,
            paddingVertical: 20,
            paddingLeft: 20,
            paddingRight: 10,
            borderRadius: 5
        }
    }
}