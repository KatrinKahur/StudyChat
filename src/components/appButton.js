import {Pressable, Text, Platform} from "react-native";
import React from "react";

export const AppButton = ({title, onPress, testID=null}) => {
    return(
        <Pressable
            testID={testID} style={styles.button}
            onPress={onPress}>
            <Text style={styles.buttonTitle}>{title}</Text>
        </Pressable>
    )
}

const styles = {
    button: {
        width: Platform.OS === 'web'? "15%" : "55%",
        minWidth: Platform.OS === 'web'? 340 : 0,
        backgroundColor: `#1e90ff`,
        paddingVertical: 17,
        marginTop: 10,
        marginBottom: 15,
        alignItems: "center",
        borderRadius: 5
    },
    buttonTitle: {
        fontSize:  Platform.OS === 'web'? 20 : 15,
        color: `#ffffff`
    }
}