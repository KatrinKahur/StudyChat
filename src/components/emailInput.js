import { Platform, StyleSheet, TextInput, View } from "react-native";
import { FontAwesome } from "@expo/vector-icons";

export const EmailInput = ({ onChangeText, onSubmitEditing, value }) => {
  return (
    <View style={styles.container}>
      <FontAwesome name="user" size={24} color="black" />
      <TextInput
          testID="emailInput"
          style={styles.input}
          value={value}
          onChangeText={onChangeText}
          onSubmitEditing={onSubmitEditing}
          placeholder="Email.."
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    marginTop: Platform.OS === "web" ? 10 : 5,
    paddingRight: 10,
    paddingTop: 20,
    paddingBottom: 5,
    borderBottomWidth: 1,
    marginVertical: 10,
  },
  input: {
    width: "100%",
    fontSize: Platform.OS === "web" ? 20 : 17,
    paddingLeft: 15,
    paddingVertical: 3,
    paddingRight: 15,
    overflow: "hidden",
  },
});
