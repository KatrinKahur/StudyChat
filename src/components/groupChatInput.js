import {TextInput, View, StyleSheet, Platform} from "react-native";
import {FontAwesome} from "@expo/vector-icons";
import React from "react";

export const GroupChatInput = ({ onChangeText, value }) => {
    return (
      <View style={styles.container}>
        <FontAwesome name="long-arrow-right" size={24} color="black" />
        <TextInput
          testID="groupChatInput"
          style={styles.input}
          value={value}
          onChangeText={onChangeText}
          placeholder="Name of chat"
        />
      </View>
    );
  };
  
  const styles = StyleSheet.create({
    container: {
      flexDirection: "row",
      marginTop: Platform.OS === "web" ? 10 : 5,
      paddingRight: 10,
      paddingTop: 20,
      paddingBottom: 5,
      borderBottomWidth: 1,
      marginVertical: 10,
    },
    input: {
      width: "100%",
      fontSize: Platform.OS === "web" ? 20 : 17,
      paddingLeft: 15,
      paddingVertical: 3,
      paddingRight: 15,
      overflow: "hidden",
    },
  });