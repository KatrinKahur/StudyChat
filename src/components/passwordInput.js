import {TextInput, View, StyleSheet, Platform} from "react-native";
import {FontAwesome5} from "@expo/vector-icons";
import React from "react";

export const PasswordInput = ({onChangeText, onSubmitEditing, value}) => {
    return(
        <View style={styles.container}>
            <FontAwesome5 name="lock" size={24} color="black" />
            <TextInput
                testID="passwordInput"
                style={styles.input} value={value}
                secureTextEntry={true}
                onSubmitEditing={onSubmitEditing}
                onChangeText = {onChangeText} placeholder="Password.."/>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        marginTop: Platform.OS === 'web'? 10 : 5,
        paddingRight: 10,
        paddingTop: 20,
        paddingBottom: 5,
        borderBottomWidth: 1,
        marginVertical: 10
    },
    input: {
        width: "100%",
        fontSize: Platform.OS === 'web'? 20 : 17,
        paddingLeft: 15,
        paddingVertical: 3,
        paddingRight: 15,
        overflow: "hidden"
    }
})