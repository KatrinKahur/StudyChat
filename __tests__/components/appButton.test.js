import renderer from "react-test-renderer";
import {AppButton} from "../../src/components/appButton";
import {cleanup, fireEvent, render} from "@testing-library/react-native";

beforeEach(() => {
    cleanup();
})

beforeEach(() => {
    global.TEST_TITLE = "Random title";
    global.TEST_ON_PRESS = () => {};
    global.TEST_ID = "testID";
});

describe('AppButton', () => {
    it('should render correctly', () => {
        const testProps = {
            title: global.TEST_TITLE,
            onPress: global.TEST_ON_PRESS,
            testID: global.TEST_ID
        }

        const { toJSON } = renderer.create(
            <AppButton {...testProps} />
        )
        expect(toJSON()).toMatchSnapshot();
    });

    it('should call the correct function when clicked on', () => {
        let changedTitle = "changed title"
        const testOnPress = () => {
            global.TEST_TITLE = changedTitle;
        }
        const testProps = {
            title: global.TEST_TITLE,
            onPress: testOnPress,
            testID: global.TEST_ID
        }

        const { getByText } = render(<AppButton {...testProps} />);
        fireEvent.press(getByText(global.TEST_TITLE));
        expect(global.TEST_TITLE).toEqual(changedTitle);
    });


    it('should set the title', () => {
        const testProps = {
            title: global.TEST_TITLE,
            onPress: global.TEST_ON_PRESS,
            testID: global.TEST_ID
        }
        const { getByText} = render(<AppButton {...testProps} />);
        expect(getByText(global.TEST_TITLE)).toBeTruthy();
    });

    it('should set the test id', () => {
        const testProps = {
            title: global.TEST_TITLE,
            onPress: global.TEST_ON_PRESS,
            testID: global.TEST_ID
        }
        const { getByTestId } = render(<AppButton {...testProps} />);
        expect(getByTestId(global.TEST_ID)).toBeTruthy();
    })
})