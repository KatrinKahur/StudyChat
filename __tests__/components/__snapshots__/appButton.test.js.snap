// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1e90ff",
      "borderRadius": 5,
      "marginBottom": 15,
      "marginTop": 10,
      "minWidth": 0,
      "paddingVertical": 17,
      "width": "55%",
    }
  }
  testID="testID"
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
      }
    }
  >
    Random title
  </Text>
</View>
`;
